crates/resolver/src/resolver.rs

```diff
@@ -80,8 +80,8 @@ impl Resolver {
         builder.enable_all();
 
         let runtime = builder.build()?;
-        let async_resolver =
-            AsyncResolver::new(config, options, TokioHandle).expect("failed to create resolver");
+        let async_resolver = AsyncResolver::new(config, options, TokioHandle::default())
```

Comment:
It seems like we really want people to pass the `TokioHandle` into this `new` method, rather than create a new one. If so, should we deprecate this `new` or add `TokioHandle` as an argument? Or do we think each Resolver should have it's own join-set?

Reply:
Each resolver needs its own JoinSet. I'm currently using this model in a proxying DNS server, where I have separate resolvers for internal vs external hostnames. If I need to tear down one resolver and recreate it, that should only terminate in-flight queries against that resolver. It should have no effect on in-flight queries against another resolver.

I don't think there's much reason to make the handle part of the signature, honestly, except I suppose to potentially support other runtimes like `async-std`. We could make `TokioHandle` private as well.

Reply:
Thanks for that answer. In that case this works as is.